### PR TITLE
Multiple variants

### LINKED ISSUE #22: Multiple variants per report
Allow entry of multiple variants per report and define the null variant data structure

- [x] convert current formik data to be an array of variants
- [x] create fhir observation for null variant
- [x] dynamically add variants to the form - Formik's [FieldArray](https://formik.org/docs/api/fieldarray) looks like it should be helpful, but would need to have this has a child of Formik component
- [x] display each variant in the confirmation page - is it easier to abstract the form display logic parts, rerendering it all together with uneditable inputs? Or maybe we just get rid of the confirmation page altogether? 
- [x] display no variant reported in the confirmation page

### PR DESCRIPTION
**Pull request by jhughes982**
- User can now add and delete individual variants,
- Will need further work on making validation work seamlessly although this can be tackled in the next PR,
- Failing tests currently being fixed in #29 , once they are fixed we can merge #29 into this PR
closes #22

### CODE REVIEW

File: src/components/reports/formSteps/Variant.tsx [hunk 1]
```diff
@@ -1,46 +1,82 @@
...
-
-      {!variantExists && <div>No variant has been set. Click the button above to set a new variant</div>}
+      <FieldArray
+        name="variant"
+        render={(arrayHelpers) => (
+          <div>
+            {values.variant &&
+              values.variant.length > 0 &&
+              values.variant.map((variant: any, index: number) => (
+                <div key={index}>
```
Comment (stefpiatek) [inline src/components/reports/formSteps/Variant.tsx:41]: ooh this works really nicely 👏

File: src/components/reports/ReportForm.tsx [hunk 2]
```diff
@@ -83,16 +82,14 @@ const ReportForm: FC<Props> = (props: Props) => {
...
     switch (formStep) {
       case 0:
         return <Patient setFieldValue={setFieldValue} />;
       case 1:
         return <Sample />;
       case 2:
-        return (
-          <Variant variantExists={variantExists} setVariantExists={setVariantExists} setFieldValue={setFieldValue} />
-        );
+        return <Variant values={values} />;
```
Comment (stefpiatek) [inline src/components/reports/ReportForm.tsx:92]: delightful slimming down of the props

**Review by stefpiatek — COMMENTED** [review]
This looks really good.

I think for this we'd need a couple of extra bits

- [ ] passing component test with no variants added
- [ ] passing component test with a variant added
- [ ] displaying of the variant in the confirmation page (from #22 )
- [ ] displaying "no variants reported" in the confirmation page (from #22 )

**Comment by jhughes982** [comment]
closes #22. Tests passing locally but failing in actions. The confirmation page required even more data manipulation to render the content, we should decide whether the confirmation page is worth keeping around and whether the current way it is rendered is satisfactory.

File: src/components/reports/formSteps/Confirmation.tsx [hunk 3]
```diff
@@ -10,11 +10,24 @@ interface Props {
...
-  const formObj = formRef.current.values;
+  let formObj = formRef.current.values;
+
+  // parse the variants and add them to the formObj as separate keys
+  if (formObj.variant && formObj.variant.length > 0) {
+    formObj.variant.map((variant: { [key: string]: string }, index: number) => {
+      formObj = { ...formObj, [`variant${index}`]: variant };
+    });
+    delete formObj.variant; // delete the old redundant variant key
+  }
```
Comment (stefpiatek) [inline src/components/reports/formSteps/Confirmation.tsx:21]: oopf yeah that is annoying that we have to do that isn't it. Agree that we should get this into users hands before we do any more on it - see how useful it is

**Review by stefpiatek — APPROVED** [review]
Chef's kiss. Very nice to see it coming together, thanks for working out the confirmation page pain.

**Pull request merged**